platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -181,11 +181,16 @@ class MGLStyleValueTransformer {
         return mbglValue;
     }
     
-    NSObject* toRawStyleSpecValue(MGLColor *color) {
-        return @(color.mgl_color.stringify().c_str());
-    }
-    
-    NSObject* toRawStyleSpecValue(NSObject *rawMGLValue) {
+    /**
+     As hack to allow converting enum => string values, we accept a second, dummy parameter in
+     the toRawStyleSpecValue() methods for converting 'atomic' (non-style-function) values.
+     This allows us to use `std::enable_if` to test (at compile time) whether or not MBGLType is an Enum.
+     */
+    template <typename MBGLEnum = MBGLType,
+    class = typename std::enable_if<!std::is_enum<MBGLEnum>::value>::type,
+    typename MGLEnum = ObjCEnum,
+    class = typename std::enable_if<!std::is_enum<MGLEnum>::value>::type>
+    NSObject* toRawStyleSpecValue(NSObject *rawMGLValue, MBGLEnum &mbglValue) {
         if ([rawMGLValue isKindOfClass:[NSValue class]]) {
             const auto rawNSValue = (NSValue *)rawMGLValue;
             if (strcmp([rawNSValue objCType], @encode(CGVector)) == 0) {
```

Comment:
This is a good catch.  I've ticketed it out [here](https://github.com/mapbox/mapbox-gl-native/issues/8088) as part of the completionist quest to propagate this approach to all of the MGL=>mbgl style property value conversions (or, at least the runtime-styling ones...)